### PR TITLE
GEODE-9987: Refactored/Removal of deprecated APIs

### PR DESCRIPTION
**Pull request by nabarunnag**
* The tests were refactored to remove deprecated functions.
	* The second test was removed as using 0 as port numbers may result in port already in use and may cause flakiness.

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

**Comment by upthewaterspout** [comment]
It's not clear to me how these changes relate to GEODE-9987. That bug was a failure saying "Member isn't responding to heartbeat requests". I don't think this refactoring is going to affect that type of failure - usually that would indicate resource issues on the CI machine or perhaps the need for a larger membership-timeout for this test? It does look like this test probably doesn't need to start so many locators, maybe that would help?

I'm not sure how I feel about removing the port 0 test. It looks like it was trying to test that members that start up with port 0 as the locator port correctly distribute the *actual* port of the locator to other members, rather than 0? Is the reason you think port 0 is going to cause port conflicts because other tests are using AvailablePortHelper? Seems like we still need a way to write tests that using port 0 actually works.

**Review by agingade — APPROVED** [review]

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java [hunk 1]
```diff
@@ -54,277 +45,163 @@
...
+@SuppressWarnings("serial")
+public class HostedLocatorsDUnitTest implements Serializable {
 
-  private String getUniqueLocatorName() {
-    String uniqueLocatorName = Host.getHost(0).getHostName() + "_"
-        + getUniqueName();
-    return uniqueLocatorName;
-  }
+  @Rule
+  public ClusterStartupRule clusterStartupRule = new ClusterStartupRule();
```
Comment (Bill) [inline geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java:52]: make private

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java [hunk 2]
```diff
@@ -54,277 +45,163 @@
...
-        @Override
-        public void run() {
-          final ClusterDistributionManager dm =
-              (ClusterDistributionManager) InternalDistributedSystem.getAnyInstance()
-                  .getDistributionManager();
-          final InternalDistributedMember self = dm.getDistributionManagerId();
-
-          final Set<InternalDistributedMember> locatorIds = dm.getLocatorDistributionManagerIds();
-          assertTrue(locatorIds.contains(self));
+  public static InetAddress getIPAddress() {
```
Comment (Bill) [inline geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java:138]: make private

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java [hunk 3]
```diff
@@ -54,277 +45,163 @@
...
-              dm.getAllHostedLocators();
-          assertTrue(!hostedLocators.isEmpty());
-          assertEquals(5, hostedLocators.size());
-
-          for (InternalDistributedMember member : hostedLocators.keySet()) {
-            assertEquals(1, hostedLocators.get(member).size());
-            final String hostedLocator = hostedLocators.get(member).iterator().next();
-            assertTrue(locators + " does not contain " + hostedLocator,
-                locators.contains(hostedLocator));
+  public static InetAddress getIPv6Address() {
```
Comment (Bill) [inline geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java:175]: make private

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java [hunk 4]
```diff
@@ -54,277 +45,163 @@
...
-          assertTrue(!hostedLocators.isEmpty());
-          assertEquals(5, hostedLocators.size());
-
-          for (InternalDistributedMember member : hostedLocators.keySet()) {
-            assertEquals(1, hostedLocators.get(member).size());
-            final String hostedLocator = hostedLocators.get(member).iterator().next();
-            assertTrue(locators + " does not contain " + hostedLocator,
-                locators.contains(hostedLocator));
+    try {
+      Enumeration i = NetworkInterface.getNetworkInterfaces();
```
Comment (Bill) [inline geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java:154]: declare as `Enumeration<NetworkInterface> i` to eliminate the cast on line 156.

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java [hunk 5]
```diff
@@ -54,277 +45,163 @@
...
+      while (i.hasMoreElements()) {
+        NetworkInterface ni = (NetworkInterface) i.nextElement();
+        Enumeration j = ni.getInetAddresses();
```
Comment (Bill) [inline geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java:157]: declare as `Enumeration<InetAddress> j` to eliminate the cast on line 159.

**Review by kirklund — APPROVED** [review]
Thanks Naba!

**Comment by nabarunnag** [comment]
All review comments have been addressed

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java (hunk 5 above)
Comment (nabarunnag) [reply geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java]: All review comments have been addressed

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java (hunk 4 above)
Comment (nabarunnag) [reply geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java]: All review comments have been addressed

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java (hunk 3 above)
Comment (nabarunnag) [reply geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java]: All review comments have been addressed

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java [hunk 6]
```diff
@@ -54,277 +45,163 @@
...
 
-          final Map<InternalDistributedMember, Collection<String>> hostedLocators =
-              dm.getAllHostedLocators();
-          assertTrue(
-              "hit bug #46324: " + hostedLocators + " is missing "
-                  + InternalLocator.getLocatorStrings() + " for " + self,
-              hostedLocators.containsKey(self));
-        }
-      });
+  protected static InetAddress getIPv4Address() {
```
[the remark this answers is not in the document]
Comment (nabarunnag) [reply geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java]: All review comments have been addressed

File: geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java (hunk 2 above)
Comment (nabarunnag) [reply geode-core/src/distributedTest/java/org/apache/geode/distributed/HostedLocatorsDUnitTest.java]: All review comments have been addressed

**Review by Bill — APPROVED** [review]
Nice one Naba! Approved!

**Pull request merged**